modules/manage/pages/cluster-maintenance/disk-utilization.adoc

```diff
@@ -56,18 +58,28 @@ Default - `null`
 | `retention.local.target.bytes`  +
 No default
 
+| Segment lifetime
+| `log_segment_ms`  +
+No default
+| `segment.ms`  +
+No default
+
 | Segment size
 | `log_segment_size`  +
 Default - `1073741824`
 | `segment.bytes`  +
 No default
 |===
 
-Data expires from object storage following `retention.ms` or `retention.bytes`. For example, if `retention.bytes` is set to 10 GiB, then every partition in the topic has a limit of 10 GiB storage in the cloud. When `retention.bytes` is exceeded by data in object storage, the data in object storage is trimmed. With Tiered Storage enabled, data expires from local storage following `retention.local.target.ms` or `retention.local.target.bytes`.
+Data expires from object storage following both `retention.ms` and `retention.bytes`. For example, if `retention.bytes` is set to 10 GiB, then every partition in the topic has a limit of 10 GiB storage. When `retention.bytes` is exceeded by data in object storage, the data in object storage is trimmed, even if `retention.ms` is not yet exceeded. With Tiered Storage enabled, data expires from local storage following `retention.local.target.ms` or `retention.local.target.bytes`.
+
+image:shared:time-based-segment-rolling.png[Time-based segment rolling]
+
+As noted above, retention policy functions by deleting or compacting closed segments. The segment lifetime and segment size configurations help ensure new segments are created regularly within each partition. The above illustration shows the relationship between segments and the retention policy using time-based limits.  If these values are set too high, especially if they're higher than the corresponding retention values, the segment may fill available disk space before closing and therefore never become eligible for deletion or compaction. If the values are set too low, your partitions will have a large number of segments which must be checked each time deletion or compaction processes execute, having a potential adverse impact on system resource utilization.
 
```

Comment:
Wanted to point out ... there's six total diagrams in the wiki covering how this works, since I had to break it down for each combination of settings to make sure it was understandable. I don't know if we want to make this a simpler diagram, or split them out to talk about each of the conditions?

Reply:
If you have a different one in mind, I'm all ears! I chose this one because it was with the docs issue and I thought it presented the information really clearly.